### ISSUE TITLE
readComposeFiles should attempt to open all files in the ComposeFiles array before returning an error

### ISSUE THREAD

**Issue by rheinwein** [docker/libcompose#388]
Given a project context, I might want to support a different filename. This is possible via `ComposeFiles`:

```
Context: project.Context{
  ComposeFiles: []string{"my-custom-compose-file.yml", "docker-compose.yml"},
  ProjectName:  "my-cool-project",
}
```

But if this project is missing `my-custom-compose-file.yml`, instead of falling back to `docker-compose.yml`, libcompose returns an error. This happens here: https://github.com/docker/libcompose/blob/4dc70801c9b8bba9c1e20ad887b6266151b7ee21/project/context.go#L60

Instead of returning an error immediately, if there are still more file paths in `c.ComposeFiles`, `readComposeFiles` should iterate through the array and only return a `not found` error if none of the files in the `ComposeFiles` array can be found.

**Comment by vdemeester** [comment]
Although I agree, this is not how `docker/compose` behave : 

```
λ docker-compose -f docker-compose.yml -f toto.yml ps
ERROR: .IOError: [Errno 2] No such file or directory: u'./toto.yml'
```

But I think this should be configurable somewhere when creating the project (hence the `kind/enhancement` label) : fail in any specified file is not found, or just if all specified file are not found.

**Comment by rheinwein** [comment]
Ah yeah. I think the current behavior makes the most sense when you are expecting "docker-compose.yml" and then an additional override file. Should specifying completely alternate filepaths (that don't adhere to the main/override pattern) go somewhere else?

**Comment by vdemeester** [comment]
> Should specifying completely alternate filepaths (that don't adhere to the main/override pattern) go somewhere else?

I don't think so. A "*don't fail too quickly on missing file*" switch is more than fine by me :stuck_out_tongue_closed_eyes:.
/cc @joshwget wdyt ?

**Comment by joshwget** [comment]
I think a switch would be fine. This could also be handled outside of libcompose pretty easily. If you have a list of filenames you want to check, you could just scan for those and pass the first one found to libcompose. Even the current behavior doesn't necessarily have to be done at the project level, but it makes sense as a default.

**Comment by rheinwein** [comment]
Yeah, I think you're right. I've already implemented a workaround outside of libcompose where we just scan for available files and then pass them to the project context. Since libcompose should have 1-1 parity with Docker Compose, I'm hesitant to introduce something else that would allow for softer error handling. Closing this for now, but would be happy to implement the switch if more people asked for it.
